src/h5web/visualizations/line/hooks.ts

```diff
@@ -4,15 +4,29 @@ import { useThree } from 'react-three-fiber';
 import { Domain } from '../shared/models';
 import { findDomain, getAxisScale } from '../shared/utils';
 import { AxisSystemContext } from '../shared/AxisSystemProvider';
-import { useVisProps } from '../../dataset-visualizer/VisProvider';
+import {
+  useVisProps,
+  useFlatValues,
+} from '../../dataset-visualizer/VisProvider';
 
 export function useData(): number[] {
-  const { rawValues, rawDims } = useVisProps();
+  const { rawValues, rawDims, mapping } = useVisProps();
+  const values = useFlatValues();
 
   if (rawDims.length === 1) {
     return rawValues;
   }
 
+  if (rawDims.length === 2) {
+    if (mapping.x === 1) {
+      return rawValues[mapping.slicingIndices[0]];
+    }
+
+    return values.filter(
+      (v, i) => i % rawDims[1] === mapping.slicingIndices[1]
+    );
```

Comment:
If we want to slice the deepest dimension, things are trickier: we have to filter the flattened array from the indices to select the appropriate values.

Unfortunately, this is not very efficient and the manipulation of the slider feels sluggish when slicing along dimension 1.
A way out would be to reorder the data at each change of the data dimension so that the slicing stays efficient.